src/api/c/random.cpp

```diff
@@ -24,24 +24,39 @@
 using namespace detail;
 using namespace common;
 
+using af::dim4;
+
+Array<uint> emptyArray()
+{
+    AF_CHECK(af_init());
```

Comment:
This should only be done at public facing APIs

Reply:
I agree too, but without initialization OpenCL backend is causing segmentation faults as we can't create Arrays without valid context/device ids.

Reply:
Probably because its missing from af_get_default_random_engine